xwiki-platform-core/xwiki-platform-container/xwiki-platform-container-servlet/src/main/java/org/xwiki/container/servlet/filters/internal/SetCharacterEncodingFilter.java

```diff
@@ -107,6 +109,8 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
                 request.setCharacterEncoding(encoding);
             }
         }
+        // Set the default encoding for the response.
```

Comment:
Could be interesting to document why we set the response to use the same encoding as the request. I don't necessarily see the relationship. You can send a request for some document but encode your request with a given encoding like ISO8859 but wan to receive the response with a UTF-8 encoding. Also shouldn't the response encodi be set only if the request encoding is set? What does it do to set a response encoding to null?

Thanks for working on this!

Reply:
As far as I understand, the parameter is mandatory and thus cannot be `null`. Also, this is supposed to be the encoding that is used by XWiki. Unless the `ignore` parameter is set, the encoding of the request is honored and the encoding of the response is set independent of the request encoding.

We probably don't really support setting any encoding but UTF-8. We could also introduce a new filter to just set the response encoding to UTF-8. It just seemed less invasive to re-use the existing filter, in particular as it name doesn't imply in any way that it only affects the request. My understanding is that in theory you can set XWiki to use another encoding than UTF-8 for everything. I don't know what happens in this case, i.e., if in this case it would be correct to use this encoding for all responses or if in this case some code would convert the content to UTF-8 at some point and the response would still need to be UTF-8. I think in practice UTF-8 is the only encoding XWiki actually supports. We should probably remove the configuration and replace all references to it by UTF-8.

Reply:
> As far as I understand, the parameter is mandatory and thus cannot be null

I've mixed `encoding` vs `this.encoding`. Would be nice to rename `encoding` as it's a bad practice (reported by SonarQube BTW).

> We probably don't really support setting any encoding but UTF-8. We could also introduce a new filter to just set the response encoding to UTF-8. It just seemed less invasive to re-use the existing filter, in particular as it name doesn't imply in any way that it only affects the request. My understanding is that in theory you can set XWiki to use another encoding than UTF-8 for everything. I don't know what happens in this case, i.e., if in this case it would be correct to use this encoding for all responses or if in this case some code would convert the content to UTF-8 at some point and the response would still need to be UTF-8. I think in practice UTF-8 is the only encoding XWiki actually supports. We should probably remove the configuration and replace all references to it by UTF-8.

I don't have problem with that but it would be nice to explain in a comment.  

Also, in parallel, it would be nice to start a discussion thread on the forum with a proposal such as "Only support UTF-8 encoding" and get an agreement there. 

Thanks